Source/mpqapi.cpp

```diff
@@ -137,58 +137,47 @@ void mpqapi_remove_hash_entry(const char *pszName)
 		block_offset = blockEntry->offset;
 		block_size = blockEntry->sizealloc;
 		memset(blockEntry, 0, sizeof(*blockEntry));
-		mpqapi_free_block(block_offset, block_size);
+		mpqapi_alloc_block(block_offset, block_size);
 		save_archive_modified = 1;
 	}
 }
 
-void mpqapi_free_block(int block_offset, int block_size)
+void mpqapi_alloc_block(int block_offset, int block_size)
 {
-	int v2;          // esi
-	int v3;          // edi
-	_BLOCKENTRY *v4; // eax
-	signed int v5;   // edx
-	signed int v6;   // ecx
-	int v7;          // ecx
-	BOOLEAN v8;      // zf
-	_BLOCKENTRY *v9; // eax
-
-	v2 = block_size;
-	v3 = block_offset;
-LABEL_2:
-	v4 = sgpBlockTbl;
-	v5 = 2048;
-	while (1) {
-		v6 = v5--;
-		if (!v6)
-			break;
-		v7 = v4->offset;
-		if (v4->offset && !v4->flags && !v4->sizefile) {
-			if (v7 + v4->sizealloc == v3) {
-				v3 = v4->offset;
-			LABEL_11:
-				v2 += v4->sizealloc;
-				memset(v4, 0, 0x10u);
-				goto LABEL_2;
+	_BLOCKENTRY *block;
+	int i;
+
+	block = sgpBlockTbl;
+	i = 2048;
+	while (i-- != 0) {
+		if (block->offset && !block->flags && !block->sizefile) {
+			if (block->offset + block->sizealloc == block_offset) {
+				block_offset = block->offset;
+				block_size += block->sizealloc;
+				memset(block, 0, sizeof(_BLOCKENTRY));
+				mpqapi_alloc_block(block_offset, block_size);
+				return;
+			}
+			if (block_offset + block_size == block->offset) {
+				block_size += block->sizealloc;
+				memset(block, 0, sizeof(_BLOCKENTRY));
+				mpqapi_alloc_block(block_offset, block_size);
+				return;
 			}
-			if (v3 + v2 == v7)
-				goto LABEL_11;
 		}
-		++v4;
+		++block;

```

Comment:
```suggestion
		block++;
```

Reply:
If its already bin exact I can't see why you would want to change it to a post increment operator instead of pre increment which better reflects what you are doing when you just increment and nothing else in a single statement.

Reply:
https://github.com/diasurgical/devilution/wiki/Code-Style

Reply:
Okay, odd style choice IMO but probably not the place to get into a discussion on it.

Reply:
Prefix `++` is more adequate choice in C++ because of semantics of its behavior for complex objects; for ints and pointers it doesn't matter performance-wise, post-fix was probably chosen as it is easier to read in general.

Reply:
It's not call ++C either ;)